### PR TITLE
OAEWDGT-151, OAEWDGT-152

### PR DESCRIPTION
**Pull request by bertpareyn**
OAEWDGT-151 Update the Widget Library FAQ -> Not complete yet but should be merged in anyway IMO.
https://jira.sakaiproject.org/browse/OAEWDGT-151

OAEWDGT-152 Create a 'GIT Basics' section
https://jira.sakaiproject.org/browse/OAEWDGT-152

### CODE REVIEW

File: app/views/sdk/faq.html.erb [hunk 1]
```diff
@@ -2,46 +2,47 @@
...
 
-    <ul class="widgetsdk_treelist">
+    <ul class="inline_list">
         <li>
-            <a href="#developmentquestions" class="wl-regular-link" title="Development Questions">Development Questions</a>
-            <ul>
-                <li><a href="#experience" class="wl-regular-link" title="What kinds of technologies do I need to know to develop Sakai OAE widgets? How much experience will I need?">What kinds of technologies do I need to know to develop Sakai OAE widgets? How much experience will I need?</a></li>
-                <li><a href="#installation" class="wl-regular-link" title="How do I get and install Sakai OAE?">How do I get and install Sakai OAE?</a></li>
-                <li><a href="#documentation" class="wl-regular-link" title="Where do I find detailed documentation on the services that are available for use in my widgets?">Where do I find detailed documentation on the services that are available for use in my widgets?</a></li>
+            <%= link_to "", "#", :class => "wl-regular-link", :title => "" %>
```
Comment (christianvuerings) [inline app/views/sdk/faq.html.erb]: Is there a need for this empty link?

File: app/views/sdk/gitbasics.html.erb [hunk 2]
```diff
@@ -0,0 +1,145 @@
+<% content_for :title do %> - SDK - GIT Basics<% end %>
+<div id="gitbasics" class="widgetsdk_container">
+    <h1>GIT Basics</h1>
+    <p>The Sakai Open Academic Environment uses GIT version control in combination with <%= link_to "Github", "https://github.com/sakaiproject", :class => "wl-regular-link", :target => "_blank" %> to manage and distribute the code. This section assumes a basic knowledge of what GIT is and what it can be used for. If you want to read up on GIT you can visit the <%= link_to "GIT homepage", "http://git-scm.com/", :title => "GIT homepage", :class => "wl-regular-link", :target => "_blank" %>.</p>
```
Comment (croby) [inline app/views/sdk/gitbasics.html.erb]: `GIT` -> `git` - it isn't an acronym, and its proper spelling is lowercase

File: app/views/sdk/gitbasics.html.erb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        <p>When setting up your environment through the <%= link_to "Full development environment setup", "/sdk/developwidget/fulldevsetup", :class => "wl-regular-link" %> you'll have one remote 'origin' which is the alias for the sakaiproject repository.</p>
+        <p>Adding remotes is as simple as the previous command, take a look at the following example where I add my own repository as a remote.</p>
+        <div class="navigationalitems_codesnippit_container few_lines">
+<pre>
+<strong>$ git remote add bp323 git@github.com:bp323/3akai-ux.git</strong>
+
+<strong>$ git remote -v</strong>
+bp323	git@github.com:bp323/3akai-ux.git (fetch)
+bp323	git@github.com:bp323/3akai-ux.git (push)
+origin	https://github.com/sakaiproject/3akai-ux.git (push)
```
Comment (croby) [inline app/views/sdk/gitbasics.html.erb]: This should be renamed to `sakaiproject` to keep in line with all our dev docs

File: app/views/sdk/gitbasics.html.erb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+<strong>$ git push bp323 :branchtodelete</strong> -> Deletes remote branch
+</pre>
+        </div>
+
+        <h2 id="commit">Committing changes</h2>
+        <p>Committing changes on a regular basis is best practice. Not only does it allow you to easily look back at fixes you did a few days (or weeks/months) ago but it also avoids losing your work when something goes horribly wrong. Also, when debugging your code it's easier to track down and remove patches if they're split up into manageable chunks. Before committing it can be useful to take a last look at the files you've edited. A handful of external applications like <%= link_to "GitX (Mac)", "http://gitx.frim.nl/", :title => "GitX homepage", :class => "wl-regular-link", :target => "_blank" %> or <%= link_to "SmartGit (Windows)", "http://www.syntevo.com/smartgit/index.html", :title => "SmartGit homepage", :class => "wl-regular-link", :target => "_blank" %> make this a graphical experience. Committing and pushing your changes is easy:</p>
+        <div class="navigationalitems_codesnippit_container">
+<pre>
+<strong>$ git add -A</strong>
+<strong>$ git commit -m "OAEWDGT-152 WIP Create a 'GIT Basics' section"</strong>
```
Comment (croby) [inline app/views/sdk/gitbasics.html.erb]: Can combine these two lines into `git commit -am`

File: app/views/sdk/faq.html.erb (hunk 1 above)
Comment (bertpareyn) [inline app/views/sdk/faq.html.erb]: Nope, leftover from the work I did in there. Removed it.

File: app/views/sdk/gitbasics.html.erb (hunk 2 above)
Comment (bertpareyn) [inline app/views/sdk/gitbasics.html.erb]: Fixed.

File: app/views/sdk/gitbasics.html.erb (hunk 3 above)
Comment (bertpareyn) [inline app/views/sdk/gitbasics.html.erb]: Fixed.

File: app/views/sdk/gitbasics.html.erb (hunk 4 above)
Comment (bertpareyn) [inline app/views/sdk/gitbasics.html.erb]: Fixed.

File: app/views/sdk/faq.html.erb [hunk 5]
```diff
@@ -67,18 +67,29 @@
...
+        <ul class="inline_list">
+            <li>Always start work on a new branch based on the master branch (keeps master clean and tidy + allows you to work on multiple widgets at once on different branches);</li>
+            <li>Committing and pushing up your work regularly is always a good idea, you don't want to lose precious code and time by having to do it over because your machine failed;</li>
+            <li>Use self-explanatory commit messages. OAE developers adopted the following commit message style: "JIRA_ID DESCRIPTION_OF_WORK". (e.g. "SAKIII-2849 Implement visual design for top navigation");</li>
+            <li>Merge your code with the master branch on a regular basis. If merge conflicts occur you can take care of them immediately instead of having to go through your code again after you've submitted a widget;</li>
+        </ul>
+        <p>If you're looking for a guide to the most useful git commands take a look at the '<%= link_to "GIT Basics", "/sdk/developwidget/gitbasics", :title => "GIT Basics", :class => "wl-regular-link" %>' section.</p>
```
Comment (christianvuerings) [inline app/views/sdk/faq.html.erb]: We should be using `Git basics` here and all the other places.

I know @croby said that the usual spelling is in lowercase, but this doesn't mean that we can't capitalize it in the beginning of a sentence or in a header (e.g. [github](http://help.github.com/mac-set-up-git/)).
Apart from that, if you look at the [other items in the menu](http://cl.ly/EzWV), we only capitalize the first word.
We should remain consistant.

Comment (bertpareyn) [inline app/views/sdk/faq.html.erb]: Fixed headers

**Comment by christianvuerings** [comment]
Squashed and merged

**Pull request closed without merging**